backend/editor/github_functions.py

```diff
@@ -135,3 +135,17 @@ async def create_pr(self, description) -> PullRequest:
                 *self.repo_info, title=title, body=body, head=self.branch_name, base="main"
             )
         ).parsed_data
+
+    async def is_pr_merged(self, pr_number: int) -> bool:
+        """
+        Check if a pull request is merged
+        """
+        try:
+            await self.connection.rest.pulls.async_check_if_merged(
+                *self.repo_info, pull_number=pr_number
+            )
+            return True
+        except RequestFailed as e:
+            if e.response.status_code == 404:
+                return False
+            raise e
```

Comment:
```suggestion
            # re-raise in case of unexpected status code
            raise e
```

Reply:
I feel comments should be reserved to logic that is difficult to understand (I'm not sure about commenting Python syntax 🤔)

Reply:
the point in the comment is the "why", here it's to specify why we re-raise.

If I propose a comment it's because reading, I have to stop and think to understand.